backend/controllers/handle_books.js

```diff
@@ -0,0 +1,145 @@
+import express from 'express';
+import fetch from 'node-fetch';
+import Bookshelf from '../models/bookshelf.js';
+import Book from '../models/book.js';
+
+const handlebooksRouter = express.Router();
+
+handlebooksRouter.post('/addBook', async (req, res) => {
+  try {
+    const { userId, bookshelfType, title, cover, author, summary, publish_date, isbn, subject, id_goodreads } = req.body;
+    const newBook = new Book({
+      title,
+      cover,
+      author,
+      summary,
+      publish_date,
+      isbn,
+      subject,
+      id_goodreads
+    });
+    const savedBook = await newBook.save();
+
+    if (userId && bookshelfType) {
+      const bookshelf = await Bookshelf.findOneAndUpdate(
+        { userId: userId, type: bookshelfType },
+        { $addToSet: { books: savedBook._id } }, // Add book to the bookshelf
+      );
+      
+      if (!bookshelf) {
+        return res.status(404).json({ message: 'Bookshelf not found and could not be created' });
+      }
+    } else {
+      return res.status(400).json({ message: 'UserId and bookshelfType are required' });
+    }
+    res.status(201).json({ message: 'Book added successfully', book: savedBook });
+  } catch (error) {
+    console.error('Error adding book:', error);
+    res.status(500).json({ message: 'Failed to add book', error: error.message });
+  }
+});
+
+handlebooksRouter.post('/moveBook', async (req, res) => {
+    try {
+        const { userId, bookId, fromShelf, toShelf } = req.body;
+    
+        await Bookshelf.updateOne(
+          { userId: userId, type: fromShelf },
+          { $pull: { books: bookId } }
+        );
+    
+        await Bookshelf.updateOne(
+          { userId: userId, type: toShelf },
+          { $addToSet: { books: bookId } }, // $addToSet prevents duplicate entries
+        );
+    
+        res.status(200).json({ message: 'Book moved successfully' });
+      } catch (error) {
+        console.error('Error moving book:', error);
+        res.status(500).json({ message: 'Failed to move book', error: error.message });
+      }
+});
+
+handlebooksRouter.get('/searchBooksName', async (req, res) => {
+  const {q:title} = req.query; // Assuming the title is passed as a query parameter
```

Comment:
We use title as the parameter here instead of q.

Reply:
i believe q is the parameter name, but title is the actual value